### ISSUE TITLE
[UX] Admin bar is missing hover delay

### ISSUE THREAD

**Issue by olafgrabienski** [backdrop/backdrop-issues#6604]
## Description of the bug

When you hover over an Admin bar menu item, the respective drop-down menu is shown immediately, without any delay. This is fine when you target the Admin bar by purpose, but it's annoying when you hit it accidentally (which can happen quite often).

## Steps To Reproduce

To reproduce the behavior:
1. Visit a Backdrop site and log in.
2. Move your mouse near the Admin bar, and move it around a bit.

## Actual behavior

As soon you hit a menu item of the Admin bar, the drop-down is opening and gets in your way.

## Expected behavior

The drop-down opens only after little delay, similar to the frontend primary navigation dropdown.

## Additional information

Add any other information that could help, such as:
- Backdrop CMS version: 1.28.0 (same for older versions)

**Comment by argiepiano** [comment]
I agree - very annoying! I run into this problem so often, especially as I usually have a custom secondary bar below the admin bar that gets obscured by the dropdown. But... is this really a bug or a feature request? Perhaps there could be another setting for "delay"?

**Comment by olafgrabienski** [comment]
> is this really a bug or a feature request?

Good question. It's however a UX issue, and those are treated as bugs. I guess when we want to add a setting (good idea), it'd be a feature request.

**Comment by olafgrabienski** [comment]
@izmeez I agree, the drop-down delay is not only relevant for the Admin bar but also for the standard (frontend) Dropdown menu. (The hamburger menu is another thing, though, as it doesn't work on hover.)

Regarding the place for a delay setting, I'd suggest to handle them per menu type:

- for the Admin bar under `admin/config/administration/admin-bar`
- for the standard Dropdown menu in the block settings

That said, I'm not sure if this issue should cover both menu types. Maybe it's better to open a follow-up for the standard Dropdown menu.

**Comment by klonos** [comment]
I have a couple of relevant thoughts on this:

1. If you have very few admins and/or content editors for a site (say, just the site owners and their developer in smaller businesses), then adding a delay would be something to agree on, and it would be easy and OK for everyone. Adding the delay somewhere that would affect all logged-in users in larger organizations is where problems start though, as some might prefer the delay, while others not. So at least for the admin bar delay, it sounds like it should be a per-user setting rather than global for the entire site.
2. Perhaps this is annoying because of the position of the admin bar. There are a lot of common tasks that require the mouse to be moved to the top of the screens, hovering over the admin bar as it "travels to its final destination". Such tasks include for instance trying to access the bookmark toolbar or the address/search bar in browsers, trying to reach the operating system menus on Mac OS etc. In that case, perhaps placing the admin bar to the side of the screen instead of having it run across the top could help (see #1842). In fact, I think that it might be a better option than the delay ...in any case, that should also be a per-user setting.

**Comment by jenlampton** [comment]
I think there's a happy medium here that does not require any settings.

We could add a very short delay that would cause the menu NOT to appear if you were to pass your cursor OVER or ACROSS the admin bar (like, on by accident, then off again right away) but still have it feel like it appears immediately once your cursor stops moving on the admin bar. 

As soon as we start adding a delay long enough that people will notice, something that would need  a per-site, per-menu, or per-user setting, that stops being something that belongs in core.

**Comment by klonos** [comment]
This issue was discussed during today's UX/Design meeting. You can watch the recording - it starts at this point: https://youtu.be/l2SAcL907AQ?t=179

**Comment by olafgrabienski** [comment]
Thanks for your suggestions, @klonos and @jenlampton! My thoughts:

Changing the Admin bar position could help with the issue, but I'd see it more as a workaround, not as a solution.

An Admin bar delay setting *per user* is nice to have at first sight, but I don't think is necessary. If the delay is reasonable it's okay to set it globally, it's comparable to make the Admin bar sticky for everyone.

Personally, I like the idea to add a very short delay (and to abstain from a setting) very much! I'm not sure if we can get consensus on a 'happy medium', but I'm looking forward to discuss a reasonable delay.

Question: Does anyone know the delay value of Backdrop's frontend Dropdown menu?

**Comment by klonos** [comment]
@olafgrabienski https://github.com/backdrop/backdrop/blob/1.x/core/misc/smartmenus/jquery.smartmenus.js#L1181 shows the following defaults:
```js
		showTimeout:		250,		// timeout before showing the sub menus
		hideTimeout:		500,		// timeout before hiding the sub menus
```

There are some other animation-related defaults there, which we might have customized though:
```js
		showDuration:		0,		// duration for show animation - set to 0 for no animation - matters only if showFunction:null
		showFunction:		null,		// custom function to use when showing a sub menu (the default is the jQuery 'show')
							// don't forget to call complete() at the end of whatever you do
							// e.g.: function($ul, complete) { $ul.fadeIn(250, complete); }
		hideDuration:		0,		// duration for hide animation - set to 0 for no animation - matters only if hideFunction:null
		hideFunction:		function($ul, complete) { $ul.fadeOut(200, complete); },	// custom function to use when hiding a sub menu (the default is the jQuery 'hide')
							// don't forget to call complete() at the end of whatever you do
							// e.g.: function($ul, complete) { $ul.fadeOut(250, complete); }
		collapsibleShowDuration:0,		// duration for show animation for collapsible sub menus - matters only if collapsibleShowFunction:null
		collapsibleShowFunction:function($ul, complete) { $ul.slideDown(200, complete); },	// custom function to use when showing a collapsible sub menu
							// (i.e. when mobile styles are used to make the sub menus collapsible)
		collapsibleHideDuration:0,		// duration for hide animation for collapsible sub menus - matters only if collapsibleHideFunction:null
		collapsibleHideFunction:function($ul, complete) { $ul.slideUp(200, complete); },	// custom function to use when hiding a collapsible sub menu
							// (i.e. when mobile styles are used to make the sub menus collapsible)
```

I hope it helps.

**Comment by klonos** [comment]
> ... which we might have customized though

A quick search in our codebase for `showTimeout` and `hideTimeout` shows that these are only used within `jquery.smartmenus.js` itself, so we might simply have the defaults.

**Comment by olafgrabienski** [comment]
>> Does anyone know the delay value of Backdrop's frontend Dropdown menu?

> showTimeout: 250
> hideTimeout: 500

Thanks for providing the values, @klonos. In my opinion, those values would be a good starting point for a "very short" Admin bar delay, as suggested by @jenlampton in https://github.com/backdrop/backdrop-issues/issues/6604#issuecomment-2166705202.

I have no idea how a delay can be added to the Admin bar. Is anyone up to provide a PR?
